### ISSUE TITLE
Build fails on Windows

### ISSUE THREAD

**Issue by alandipert** [rstudio/httpuv#182]
I tried to install `shinyloadtest` but it failed when trying to build `httpuv` from github. Calling `devtools::install_github("rstudio/httpuv")` appears to fail in the same way.

In the build output below, the most relevant error to me looked like `make[1]: *** No rule to make target 'Makefile.mingw'`. I see that in `src/Makevars.win:24` `make` is called with `Makefile.mingw`, but that file doesn't appear to exist in this repo.

Session info:
```
- Session info ---------------------------------------------------------------------------
 setting  value                       
 version  R version 3.5.1 (2018-07-02)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       America/Los_Angeles         
 date     2018-12-18                  

- Packages -------------------------------------------------------------------------------
 package     * version date       lib source        
 assertthat    0.2.0   2017-04-11 [1] CRAN (R 3.5.1)
 backports     1.1.2   2017-12-13 [1] CRAN (R 3.5.0)
 callr         3.1.0   2018-12-10 [1] CRAN (R 3.5.1)
 cli           1.0.1   2018-09-25 [1] CRAN (R 3.5.1)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 3.5.1)
 curl          3.2     2018-03-28 [1] CRAN (R 3.5.1)
 desc          1.2.0   2018-05-01 [1] CRAN (R 3.5.1)
 devtools      2.0.1   2018-10-26 [1] CRAN (R 3.5.1)
 digest        0.6.18  2018-10-10 [1] CRAN (R 3.5.1)
 fs            1.2.6   2018-08-23 [1] CRAN (R 3.5.1)
 glue          1.3.0   2018-07-17 [1] CRAN (R 3.5.1)
 magrittr      1.5     2014-11-22 [1] CRAN (R 3.5.1)
 memoise       1.1.0   2017-04-21 [1] CRAN (R 3.5.1)
 pkgbuild      1.0.2   2018-10-16 [1] CRAN (R 3.5.1)
 pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.1)
 prettyunits   1.0.2   2015-07-13 [1] CRAN (R 3.5.1)
 processx      3.2.1   2018-12-05 [1] CRAN (R 3.5.1)
 ps            1.2.1   2018-11-06 [1] CRAN (R 3.5.1)
 R6            2.3.0   2018-10-04 [1] CRAN (R 3.5.1)
 Rcpp          1.0.0   2018-11-07 [1] CRAN (R 3.5.1)
 remotes       2.0.2   2018-10-30 [1] CRAN (R 3.5.1)
 rlang         0.3.0.1 2018-10-25 [1] CRAN (R 3.5.1)
 rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.5.1)
 rstudioapi    0.8     2018-10-02 [1] CRAN (R 3.5.1)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.1)
 usethis       1.4.0   2018-08-14 [1] CRAN (R 3.5.1)
 withr         2.1.2   2018-03-15 [1] CRAN (R 3.5.1)

[1] C:/Users/aland/R/win-library/3.5
[2] C:/Program Files/R/R-3.5.1/library
```
Build output:
```
> devtools::install_github('rstudio/shinyloadtest')
Downloading GitHub repo rstudio/shinyloadtest@master
Downloading GitHub repo rstudio/httpuv@master
√  checking for file 'C:\Users\aland\AppData\Local\Temp\Rtmpq6LVe8\remotese383f495003\rstudio-httpuv-30003ae/DESCRIPTION' (571ms)
-  preparing 'httpuv': (3.2s)
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts (506ms)
-  checking for empty or unneeded directories (3.5s)
-  building 'httpuv_1.4.5.9002.tar.gz'
   Warning: file 'httpuv/src/libuv/configure' did not have execute permissions: corrected
   
Installing package into ‘C:/Users/aland/R/win-library/3.5’
(as ‘lib’ is unspecified)
* installing *source* package 'httpuv' ...
** libs

*** arch - i386
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c RcppExports-legacy.cpp -o RcppExports-legacy.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c RcppExports.cpp -o RcppExports.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c callback.cpp -o callback.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c callbackqueue.cpp -o callbackqueue.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c filedatasource-unix.cpp -o filedatasource-unix.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c filedatasource-win.cpp -o filedatasource-win.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c http.cpp -o http.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c httprequest.cpp -o httprequest.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c httpresponse.cpp -o httpresponse.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c httpuv.cpp -o httpuv.o
C:/Rtools/mingw_32/bin/gcc  -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O3 -Wall  -std=gnu99 -mtune=generic -c md5.c -o md5.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c mime.cpp -o mime.o
C:/Rtools/mingw_32/bin/gcc  -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O3 -Wall  -std=gnu99 -mtune=generic -c set_invalid_parameter.c -o set_invalid_parameter.o
set_invalid_parameter.c:9:28: warning: '_set_invalid_parameter_handler' redeclared without dllimport attribute: previous dllimport ignored [-Wattributes]
 _invalid_parameter_handler _set_invalid_parameter_handler(_invalid_parameter_handler pNew) {
                            ^
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c socket.cpp -o socket.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c staticpath.cpp -o staticpath.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c thread.cpp -o thread.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c utils.cpp -o utils.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c uvutil.cpp -o uvutil.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c webapplication.cpp -o webapplication.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c websockets-base.cpp -o websockets-base.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c websockets-hixie76.cpp -o websockets-hixie76.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c websockets-hybi03.cpp -o websockets-hybi03.o
websockets-hybi03.cpp: In member function 'virtual void WebSocketProto_HyBi03::handshake(const string&, const RequestHeaders&, char**, size_t*, ResponseHeaders*, std::vector<unsigned char>*) const':
websockets-hybi03.cpp:66:41: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   *reinterpret_cast<uint32_t*>(handshake) = key1;
                                         ^
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c websockets-ietf.cpp -o websockets-ietf.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-35~1.1/include" -DNDEBUG -Ilibuv/include -D_WIN32_WINNT=0x0600 -DSTRICT_R_HEADERS -Wno-deprecated -I"C:/Users/aland/R/win-library/3.5/Rcpp/include" -I"C:/Users/aland/R/win-library/3.5/BH/include" -I"C:/Users/aland/R/win-library/3.5/later/include"        -O2 -Wall  -mtune=generic -c websockets.cpp -o websockets.o
/Rtools/bin/make --directory=libuv -f Makefile.mingw \
	CC="C:/Rtools/mingw_32/bin/gcc " CFLAGS="-O3 -Wall  -std=gnu99 -mtune=generic -Iinclude -Isrc -Isrc/win -DWIN32_LEAN_AND_MEAN -D_WIN32_WINNT=0x0600  " \
	AR="C:/Rtools/mingw_32/bin/ar" RANLIB="C:/Rtools/mingw_32/bin/ranlib" HAVE_DTRACE=0
make[1]: Entering directory '/Users/aland/AppData/Local/Temp/RtmpuURD2S/R.INSTALL109c525b6c29/httpuv/src-i386/libuv'
make[1]: Makefile.mingw: No such file or directory
make[1]: *** No rule to make target 'Makefile.mingw'.  Stop.
make[1]: Leaving directory '/Users/aland/AppData/Local/Temp/RtmpuURD2S/R.INSTALL109c525b6c29/httpuv/src-i386/libuv'
make: *** [Makevars.win:24: libuv/libuv.a] Error 2
ERROR: compilation failed for package 'httpuv'
* removing 'C:/Users/aland/R/win-library/3.5/httpuv'
In R CMD INSTALL
Error in i.p(...) : 
  (converted from warning) installation of package ‘C:/Users/aland/AppData/Local/Temp/Rtmpq6LVe8/filee38a31d69/httpuv_1.4.5.9002.tar.gz’ had non-zero exit status
```

**Comment by wch** [comment]
This is probably related to an update to libuv that we made in httpuv, which (I think) incorporated this change:

https://github.com/libuv/libuv/commit/3045fe1242ca51149507d3fecdadf6a8d44edcd1

I think the solution will be to copy over the libuv-related make targets from src/Makevars to src/Makevars.win, making sure to keep whatever Windows customizations are necessary.

**Comment by wch** [comment]
FWIW, I don't think that shinyloadtest needs to install httpuv from github anymore. I think you can remove it from the `Remotes` section of shinyloadtest's DESCRIPTION file.

**Comment by alandipert** [comment]
Thanks, yes I was able to get shinyloadtest installed and working by obtaining from CRAN instead.

**Comment by wch** [comment]
This was fixed by #183.
